### PR TITLE
add Graph Test for Track 2 Test Migration

### PR DESCRIPTION
**Pull request by nisha-bhatia**
Also update Cassandra db account name

### CODE REVIEW

File: sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/SessionRecords/CassandraResourcesOperationsTests/CassandraCRUDTestsAsync()Async.json [hunk 1]
```diff
@@ -4,7 +4,6 @@
       "RequestUri": "https://management.azure.com/subscriptions/db1ab6f0-4769-4b27-930e-01e2ef9c123c/resourcegroups/Default-CosmosDB-8918?api-version=2017-05-10",
       "RequestMethod": "PUT",
       "RequestHeaders": {
-        "Accept": "application/json",
```
Comment (bquantump) [inline sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/SessionRecords/CassandraResourcesOperationsTests/CassandraCRUDTestsAsync()Async.json]: Did you manually do this or merged master and regenerated?

File: sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/ScenarioTests/CassandraResourcesOperationsTests.cs [hunk 2]
```diff
@@ -33,6 +32,7 @@ public async Task ClearAndInitialize()
             {
                 InitializeClients();
                 resourceGroupName = Recording.GenerateAssetName(CosmosDBTestUtilities.ResourceGroupPrefix);
+                databaseAccountName = Recording.GenerateAssetName("amecassandratest");
```
Comment (bquantump) [inline sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/ScenarioTests/CassandraResourcesOperationsTests.cs:35]: Entropy of this is still low, but we will change this soon anyway

File: sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/ScenarioTests/GraphResourcesOperationsTests.cs [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+            var gremlinPageableResultsTask = CosmosDBManagementClient.GremlinResources.ListGremlinGraphsAsync(resourceGroupName, databaseAccountName, databaseName).ToEnumerableAsync();
+            List<GremlinGraphGetResults> gremlinGraphs = gremlinPageableResultsTask.ConfigureAwait(false).GetAwaiter().GetResult();
+            Assert.NotNull(gremlinGraphs);
+        }
+
+        private void VerifyGremlinGraphCreation(GremlinGraphGetResults gremlinGraphGetResults, GremlinGraphCreateUpdateParameters gremlinGraphCreateUpdateParameters)
+        {
+            Assert.AreEqual(gremlinGraphGetResults.Resource.Id, gremlinGraphCreateUpdateParameters.Resource.Id);
+            Assert.AreEqual(gremlinGraphGetResults.Resource.IndexingPolicy.IndexingMode.Value.ToString().ToLower(), gremlinGraphCreateUpdateParameters.Resource.IndexingPolicy.IndexingMode.Value.ToString().ToLower());
+            //Assert.AreEqual(gremlinGraphGetResults.Resource.IndexingPolicy.ExcludedPaths, gremlinGraphCreateUpdateParameters.Resource.IndexingPolicy.ExcludedPaths);
```
Comment (bquantump) [inline sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/ScenarioTests/GraphResourcesOperationsTests.cs:140]: Do not have comment out code in production code.

File: sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/ScenarioTests/GraphResourcesOperationsTests.cs [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+            locations.Add(loc);
+
+            DatabaseAccountCreateUpdateParameters databaseAccountCreateUpdateParameters = new DatabaseAccountCreateUpdateParameters(locations);
+            databaseAccountCreateUpdateParameters.Location = CosmosDBTestUtilities.Location;
+            databaseAccountCreateUpdateParameters.Capabilities.Add(new Capability("EnableGremlin"));
+            databaseAccountCreateUpdateParameters.ConsistencyPolicy = new ConsistencyPolicy(DefaultConsistencyLevel.Eventual);
+
+            await WaitForCompletionAsync(await CosmosDBManagementClient.DatabaseAccounts.StartCreateOrUpdateAsync(resourceGroupName, databaseAccountName, databaseAccountCreateUpdateParameters));
+
+            Task<Response> taskIsDatabaseNameExists = CosmosDBManagementClient.DatabaseAccounts.CheckNameExistsAsync(databaseAccountName);
```
Comment (bquantump) [inline sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/ScenarioTests/GraphResourcesOperationsTests.cs:65]: Just use await here, then you won't need to do all these extra steps you are doing

**Review by bquantump — CHANGES_REQUESTED** [review]
Respond to comments or fix them

File: sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/ScenarioTests/GraphResourcesOperationsTests.cs [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+            Response isDatabaseNameExists = taskIsDatabaseNameExists.ConfigureAwait(false).GetAwaiter().GetResult();
+            Assert.AreEqual(200, isDatabaseNameExists.Status);
+
+            GremlinDatabaseCreateUpdateParameters gremlinDatabaseCreateUpdateParameters = new GremlinDatabaseCreateUpdateParameters(new GremlinDatabaseResource(databaseName), new CreateUpdateOptions());
+            var gremlinDatabaseResponse = await WaitForCompletionAsync(await CosmosDBManagementClient.GremlinResources.StartCreateUpdateGremlinDatabaseAsync(resourceGroupName, databaseAccountName, databaseName, gremlinDatabaseCreateUpdateParameters));
+            GremlinDatabaseGetResults gremlinDatabaseGetResults = gremlinDatabaseResponse.Value;
+            Assert.NotNull(gremlinDatabaseGetResults);
```
Comment (bquantump) [inline sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/ScenarioTests/GraphResourcesOperationsTests.cs:72]: Assert the response is not null before assert the value of the response is not null

File: sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/ScenarioTests/GraphResourcesOperationsTests.cs [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+            Assert.AreEqual(databaseName, gremlinDatabaseGetResults.Name);
+
+            var gremlinDatabaseResponse1 = await WaitForCompletionAsync(await CosmosDBManagementClient.GremlinResources.StartCreateUpdateGremlinDatabaseAsync(resourceGroupName, databaseAccountName, databaseName, gremlinDatabaseCreateUpdateParameters));
+            GremlinDatabaseGetResults gremlinDatabaseGetResults1 = gremlinDatabaseResponse1.Value;
+            Assert.NotNull(gremlinDatabaseGetResults);
```
Comment (bquantump) [inline sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/ScenarioTests/GraphResourcesOperationsTests.cs:77]: Assert the response is not null before assert the value of the response is not null

File: sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/ScenarioTests/GraphResourcesOperationsTests.cs [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+            Assert.AreEqual(databaseName, gremlinDatabaseGetResults1.Name);
+
+            VerifyEqualGremlinDatabases(gremlinDatabaseGetResults, gremlinDatabaseGetResults1);
+
+            GremlinDatabaseCreateUpdateParameters gremlinDatabaseCreateUpdateParameters2 = new GremlinDatabaseCreateUpdateParameters(new GremlinDatabaseResource(databaseName2), new CreateUpdateOptions(sampleThroughput, default));
+            var gremlinDatabaseResponse2 = await WaitForCompletionAsync(await CosmosDBManagementClient.GremlinResources.StartCreateUpdateGremlinDatabaseAsync(resourceGroupName, databaseAccountName, databaseName2, gremlinDatabaseCreateUpdateParameters2));
+            GremlinDatabaseGetResults gremlinDatabaseGetResults2 = gremlinDatabaseResponse2.Value;
+
+            Assert.NotNull(gremlinDatabaseGetResults2);
```
Comment (bquantump) [inline sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/ScenarioTests/GraphResourcesOperationsTests.cs:86]: Assert the response is not null before assert the value of the response is not null

File: sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/ScenarioTests/GraphResourcesOperationsTests.cs [hunk 8]
```diff
@@ -0,0 +1,162 @@
...
+            Assert.AreEqual(databaseName2, gremlinDatabaseGetResults2.Name);
+
+            var gremlinResponseTask = CosmosDBManagementClient.GremlinResources.ListGremlinDatabasesAsync(resourceGroupName, databaseAccountName).ToEnumerableAsync();
```
Comment (bquantump) [inline sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/ScenarioTests/GraphResourcesOperationsTests.cs:89]: Just await here

File: sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/ScenarioTests/GraphResourcesOperationsTests.cs [hunk 9]
```diff
@@ -0,0 +1,162 @@
...
+            List<GremlinDatabaseGetResults> gremlinDatabases = gremlinResponseTask.ConfigureAwait(false).GetAwaiter().GetResult();
+            Assert.NotNull(gremlinDatabases);
+
+            var throughputResponse = CosmosDBManagementClient.GremlinResources.GetGremlinDatabaseThroughputAsync(resourceGroupName, databaseAccountName, databaseName2);
```
Comment (bquantump) [inline sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/ScenarioTests/GraphResourcesOperationsTests.cs:93]: await here

File: sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/SessionRecords/CassandraResourcesOperationsTests/CassandraCRUDTestsAsync()Async.json (hunk 1 above)
Comment (nisha-bhatia) [reply sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/SessionRecords/CassandraResourcesOperationsTests/CassandraCRUDTestsAsync()Async.json]: Regenerated

File: sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/ScenarioTests/GraphResourcesOperationsTests.cs [hunk 10]
```diff
@@ -0,0 +1,162 @@
...
+            uniqueKeys.Add(uk);
+            UniqueKeyPolicy uniqueKeyPolicy = new UniqueKeyPolicy(uniqueKeys);
+
+            ConflictResolutionPolicy conflictResolutionPolicy = new ConflictResolutionPolicy(new ConflictResolutionMode("LastWriterWins"), "/path", "");
+
+            GremlinGraphCreateUpdateParameters gremlinGraphCreateUpdateParameters = new GremlinGraphCreateUpdateParameters(new GremlinGraphResource(gremlinGraphName, indexingPolicy, containerPartitionKey, -1, uniqueKeyPolicy, conflictResolutionPolicy), new CreateUpdateOptions(sampleThroughput, default));
+
+            Response<GremlinGraphGetResults> gremlinResponse = await WaitForCompletionAsync(await CosmosDBManagementClient.GremlinResources.StartCreateUpdateGremlinGraphAsync(resourceGroupName, databaseAccountName, databaseName, gremlinGraphName, gremlinGraphCreateUpdateParameters));
+            GremlinGraphGetResults gremlinGraphGetResults = gremlinResponse.Value;
+            Assert.NotNull(gremlinGraphGetResults);
```
Comment (bquantump) [inline sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/ScenarioTests/GraphResourcesOperationsTests.cs:127]: Assert the response is not null before assert the value of the response is not null

File: sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/ScenarioTests/GraphResourcesOperationsTests.cs [hunk 11]
```diff
@@ -0,0 +1,162 @@
...
+
+            VerifyGremlinGraphCreation(gremlinGraphGetResults, gremlinGraphCreateUpdateParameters);
+
+            var gremlinPageableResultsTask = CosmosDBManagementClient.GremlinResources.ListGremlinGraphsAsync(resourceGroupName, databaseAccountName, databaseName).ToEnumerableAsync();
```
Comment (bquantump) [inline sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/ScenarioTests/GraphResourcesOperationsTests.cs:131]: await here

File: sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/ScenarioTests/GraphResourcesOperationsTests.cs (hunk 3 above)
Comment (nisha-bhatia) [reply sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/ScenarioTests/GraphResourcesOperationsTests.cs:140]: Removed

File: sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/ScenarioTests/GraphResourcesOperationsTests.cs [hunk 12]
```diff
@@ -0,0 +1,162 @@
...
+            List<GremlinGraphGetResults> gremlinGraphs = gremlinPageableResultsTask.ConfigureAwait(false).GetAwaiter().GetResult();
+            Assert.NotNull(gremlinGraphs);
```
Comment (bquantump) [inline sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/ScenarioTests/GraphResourcesOperationsTests.cs:133]: Should do more than just asserting not null here, should really test that the list is what you expect it to be

**Comment by bquantump** [comment]
There is too much going on in this test, you should consider splitting list into separate tests. There is no way to just test list for example without running this entire test before. One should be able to do dotnet test --filter TestList or something

**Comment by YalinLi0312** [comment]
Just curious, I don't see any changes in OperationsTest, why you update its recording?

File: sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/ScenarioTests/CassandraResourcesOperationsTests.cs (hunk 2 above)
Comment (YalinLi0312) [reply sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/ScenarioTests/CassandraResourcesOperationsTests.cs:35]: I will update all random name generations in CosmosDB after I finished the random generator in the test framework.

**Comment by nisha-bhatia** [comment]
I did not mean to update the recording, I re-generated it but have restored it to it's previous values, but I can just remove the file.

File: sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/ScenarioTests/GraphResourcesOperationsTests.cs (hunk 9 above)
Comment (nisha-bhatia) [reply sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/ScenarioTests/GraphResourcesOperationsTests.cs]: resolved

File: sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/ScenarioTests/GraphResourcesOperationsTests.cs (hunk 4 above)
Comment (nisha-bhatia) [reply sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/ScenarioTests/GraphResourcesOperationsTests.cs]: resolved

File: sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/ScenarioTests/GraphResourcesOperationsTests.cs (hunk 5 above)
Comment (nisha-bhatia) [reply sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/ScenarioTests/GraphResourcesOperationsTests.cs:72]: Updated

File: sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/ScenarioTests/GraphResourcesOperationsTests.cs (hunk 6 above)
Comment (nisha-bhatia) [reply sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/ScenarioTests/GraphResourcesOperationsTests.cs:78]: Updated

File: sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/ScenarioTests/GraphResourcesOperationsTests.cs (hunk 7 above)
Comment (nisha-bhatia) [reply sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/ScenarioTests/GraphResourcesOperationsTests.cs:87]: Updated

File: sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/ScenarioTests/GraphResourcesOperationsTests.cs (hunk 8 above)
Comment (nisha-bhatia) [reply sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/ScenarioTests/GraphResourcesOperationsTests.cs]: resolved

File: sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/ScenarioTests/GraphResourcesOperationsTests.cs (hunk 10 above)
Comment (nisha-bhatia) [reply sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/ScenarioTests/GraphResourcesOperationsTests.cs:127]: Resolved

File: sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/ScenarioTests/GraphResourcesOperationsTests.cs (hunk 11 above)
Comment (nisha-bhatia) [reply sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/ScenarioTests/GraphResourcesOperationsTests.cs]: resolved

File: sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/ScenarioTests/GraphResourcesOperationsTests.cs (hunk 12 above)
Comment (nisha-bhatia) [reply sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/ScenarioTests/GraphResourcesOperationsTests.cs:132]: I test that the graph is created correctly in the helper function called on Line 129. I have added code to call the same function on the graph that we get back to verify that it is created correctly.

**Review by YalinLi0312 — CHANGES_REQUESTED** [review]

**Review by bquantump — APPROVED** [review]

File: sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/ScenarioTests/GraphResourcesOperationsTests.cs [hunk 13]
```diff
@@ -0,0 +1,197 @@
...
+            Response<GremlinGraphGetResults> gremlinResponse = await WaitForCompletionAsync(await CosmosDBManagementClient.GremlinResources.StartCreateUpdateGremlinGraphAsync(resourceGroupName, databaseAccountName, databaseName, gremlinGraphName, gremlinGraphCreateUpdateParameters));
+            Assert.NotNull(gremlinResponse);
+            GremlinGraphGetResults gremlinGraphGetResults = gremlinResponse.Value;
+            Assert.NotNull(gremlinGraphGetResults);
+
+            VerifyGremlinGraphCreation(gremlinGraphGetResults, gremlinGraphCreateUpdateParameters);
+
+            List<GremlinGraphGetResults> gremlinGraphs = await CosmosDBManagementClient.GremlinResources.ListGremlinGraphsAsync(resourceGroupName, databaseAccountName, databaseName).ToEnumerableAsync();
+            Assert.NotNull(gremlinGraphs);
+        }
```
[the remark this answers is not in the document]
Comment (bquantump) [reply sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/ScenarioTests/GraphResourcesOperationsTests.cs:170]: This fixed:

https://github.com/Azure/azure-sdk-for-net/blob/432cffec25dd9d53e02102b69b7d15c2b695bb34/sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/ScenarioTests/GraphResourcesOperationsTests.cs#L115

https://github.com/Azure/azure-sdk-for-net/blob/432cffec25dd9d53e02102b69b7d15c2b695bb34/sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/ScenarioTests/GraphResourcesOperationsTests.cs#L125

**Review by bquantump — CHANGES_REQUESTED** [review]
Shorten long lines, then it should be good

**Review by YalinLi0312 — COMMENTED** [review]
Looks good to me. One minor comment, you may remove the get throughput tests by accident.

**Comment by nisha-bhatia** [comment]
That test got renamed to UpdateThroughput so it’s still there

**Comment by YalinLi0312** [comment]
I searched "GetCassandraKeyspaceThroughputAsync", nothing found

**Comment by nisha-bhatia** [comment]
I use this function now: .StartUpdateGremlinGraphThroughputAsync

**Review by bquantump — APPROVED** [review]

**Review by YalinLi0312 — APPROVED** [review]

**Pull request merged**